draft-ajitomi-cose-cose-key-jwk-hpke-kem.md

```diff
@@ -120,17 +107,9 @@ HPKE_Key_Configuration = [
 
 The hkc parameter can be used with existing OKP and EC2 keys {{RFC9053}} and the keys for future post-quantum KEMs.
 
-### Restrictions on the Use of Existing Key Parameters
-
-The restrictions on the use of existing common key parameters in a COSE_Key for the HPKE KEM are as follows:
-
-- alg(3): The parameter MUST be present and contains one of the following values:
-  - HPKE-v1-Base (T.B.D.)
-  - HPKE-v1-PSK (T.B.D.)
-  - HPKE-v1-Auth (T.B.D.)
-  - HPKE-v1-AuthPSK (T.B.D.)
-- key_ops(4): The parameter SHOULD NOT be specified. If specified, it MUST include "derive key"(7) and/or "derive bits"(8).
-- etc.
```

Comment:
@laurencelundblade Thank you for the pull request! I agree with the part you added below, but I cannot understand the necessity of removing the section above. I think this section has no conflict with the part you added and I also think we have to mention these kind of restrictions because RFC9052 allows us to use these common key parameters (alg, key_ops, etc.) for all key types. Could you please tell me the reason?

```
When this is present, it restricts restricts use of the key to HPKE and to HPKE only with the listed algorithms.
This structure is essentially extra detail for the "alg" parameter specified in section 7.1 of [RFC9052] and thus carries the same semantics as it.
This extra detail is needed because COSE-HPKE doesn't use the ciphersuite basis for defining algorithms that most of COSE does.
```